src/core/model_config_utils.cc

```diff
@@ -637,8 +637,15 @@ GetNormalizedModelConfig(
 {
   // If 'autofill' then the configuration file can be empty.
   const auto config_path = JoinPath({path, kModelConfigPbTxt});
+
+  // Cloud file systems may report error when config file is missing. As a
+  // workaround, we assume that an error indicates that the file is missing.
```

Comment:
Shouldn't we improve `FileExists()` function for cloud file systems that it sets `model_config_exists` and return Status::Success if the file is missing? So when `FileExists()` does return error (i.e. can't connect to storage) and we will report it instead of silently ignoring it.

Reply:
In `S3FileSystem::FileExists()`
For the case when `head_object_outcome.IsSuccess()` is false in the filesystem we cannot know if it is due to a credential 
 or access error or if the file itself is missing. Due to this reason the handling for the model config when autofill is enabled with be different since we can proceed with the assumption that the config file is missing. 
 
 The same is also true for GCS / Azure. 
 
 We can log the error that we were unable to access the <config file path> if you think that is necessary.

Reply:
`head_object_outcome` doesn't have a corresponding error code for "file not found"? In either case, I think at least the logic you have here should be moved to `FileExists()` function so the [interface](https://github.com/triton-inference-server/server/blob/main/src/core/filesystem.h#L89-L92) is consistent: `exists` is set to whether the file exists and error is returned if unable to perform the check. Otherwise your change here implies returning error also means the file doe not exist..

Reply:
Found something for S3, looking for similar error codes for GCS and Azure